### ISSUE TITLE
WriteConfig fails to write if file does not exist

### ISSUE THREAD

**Issue by lllama** [spf13/viper#433]
The new `WriteConfig` function ends up calling `findConfigFile`, which in turn checks whether the file exists. If it doesn't then it returns an error, which stops the config being written.

**Comment by michaeleekk** [comment]
I had similar issue too and filed as #430 and right now my workaround is like this, 

```
if err := viper.SafeWriteConfigAs(configPath); err != nil {
	if os.IsNotExist(err) {
		err = viper.WriteConfigAs(configPath)
	}
}
```

I'm just hoping there is a better solution.

**Comment by parkervcp** [comment]
Use `WriteConfigAs` which creates the file before writing.

**Comment by koooge** [comment]
I got same issue.
I think WriteConfig should create configfile as well as WriteConfigAs

**Comment by mloskot** [comment]
@parkervcp 
> Use `WriteConfigAs` which creates the file before writing.

Nope, it does not. So `WriteConfigAs` in https://github.com/spf13/viper/issues/433#issuecomment-356483379 will fail if directory does not exist where the config file is supposed to be written.
